front/src/common/Pathfinding/TypeAndPath.tsx

```diff
@@ -0,0 +1,179 @@
+};
+
+const monospaceOneCharREMWidth = 0.6225;
+
+function OpTooltips({ opList }: { opList: SearchResultItemOperationalPoint[] }) {
+  // Calculation of chars distance from left to put tooltip on center of op name
+  const calcLeftMargin = (charsFromLeft: number, length: number) =>
+    charsFromLeft * monospaceOneCharREMWidth + (length * monospaceOneCharREMWidth) / 2;
+  let charsFromLeft = 0;
+  return (
+    <div className="op-tooltips">
+      {opList.map((op, idx) => {
+        const leftMargin = calcLeftMargin(charsFromLeft, op.trigram.length);
+        charsFromLeft = charsFromLeft + op.trigram.length + 1;
+        return (
+          op.trigram !== '' && (
+            <div
+              className={cx('op', { wrong: !op.name })}
+              key={`typeandpath-op-${idx}-${op.trigram}`}
+              style={{ left: `${leftMargin}rem` }}
+            >
+              {op.name ? op.name : <GoAlert />}
+            </div>
+          )
+        );
+      })}
+    </div>
+  );
+}
+
+export default function TypeAndPath({ zoomToFeature }: PathfindingProps) {
+  const dispatch = useDispatch();
+  const [inputText, setInputText] = useState<string>('');
+  const [opList, setOpList] = useState<SearchResultItemOperationalPoint[]>([]);
+  const infraId = useSelector(getInfraID);
+  const rollingStockId = useSelector(getRollingStockID);
+  const [postSearch] = osrdEditoastApi.usePostSearchMutation();
+  const [postPathfinding] = osrdEditoastApi.usePostPathfindingMutation();
+  const { t } = useTranslation('operationalStudies/manageTrainSchedule');
+
+  const debouncedInputText = useDebounce(inputText.trimEnd(), 500);
+
+  const handleInput = (text: string) => {
+    // setInputText(text.trimStart().toUpperCase());
+    setInputText(text.trimStart());
+  };
+
+  function getOpNames() {
+    if (infraId !== undefined) {
+      const opTrigrams = inputText.toUpperCase().split(' ');
+      const constraint = opTrigrams.reduce(
+        (res, trigram) => [...res, ['=', ['trigram'], trigram]],
+        ['or'] as (string | SearchConstraintType)[]
+      );
+      const limitToMainStationConstraint = [
+        'or',
+        ['=', ['ch'], ''],
+        ['=', ['ch'], 'BV'],
+        ['=', ['ch'], '00'],
+      ];
+      const payload: PostSearchApiArg = {
+        searchPayload: {
+          object: 'operationalpoint',
+          query: ['and', constraint, ['=', ['infra_id'], infraId], limitToMainStationConstraint],
+        },
+        pageSize: 1000,
+      };
+      postSearch(payload)
+        .unwrap()
+        .then((results) => {
+          const operationalPoints = [...results] as SearchResultItemOperationalPoint[];
+          setOpList(
+            opTrigrams.map(
+              (trigram) => operationalPoints.find((op) => op.trigram === trigram) || { trigram }
+            ) as SearchResultItemOperationalPoint[]
+          );
+        });
+    }
+  }
+
+  const isInvalid = !opList.every((op) => op.name || op.trigram === '');
```

Comment:
I prefer the `some` method because as soon as a condition is true, the loop ends.
Also the useMemo will prevent undesired computation when the component rerenders.
```suggestion
  const isInvalid= useMemo(() => {
    opList.some((op) => !op.name && op.trigram !== '');
  }, [opList]);
```
Should the button be also disabled if the input is empty ?

Reply:
done ! including for button